SofaKernel/framework/sofa/helper/system/thread/CircularQueue.inl

```diff
@@ -295,13 +298,18 @@ inline SOFA_HELPER_API bool ManyThreadsPerEnd::push(AtomicInt array[], int maxSi
     }
 
     // atomically reserve the element to write
-    int writeIdx = tail.exchange_and_add(1) & (maxCapacity-1); // maxCapacity is assumed to be a power of 2
+    int writeIdx = tail.fetch_add(1) & (maxCapacity-1); // maxCapacity is assumed to be a power of 2
     // Active wait:
     // loop as long as the element has not been read by another thread (which is indicated by a -1 value).
     // It happens when the queue is temporarily full.
-    while(array[writeIdx].compare_and_swap(-1, item) != -1)
+
+    int refInt;
+    do
     {
+        refInt = -1;
     }
+    while(!array[writeIdx].compare_exchange_strong(refInt, item) );
+    //while(array[writeIdx].compare_and_swap(-1, item) != -1)
```

Comment:
Keeping the old line on purpose?

Reply:
Yes, I am not so sure that what I have done what the correct thing to do; so I kept the old line to check the previous behavior....